torch/_inductor/codegen/wrapper.py

```diff
@@ -1018,6 +1023,58 @@ def write_header(self):
                 """
             )
 
+    def write_inf_and_nan_header(self):
+        if self.inf_and_nan_header:
+            return
+
+        self.inf_and_nan_header = True
+        self.header.splice(
+            """
+            static inline bool check_has_inf(at::Tensor tensor, bool assertion) {
+              bool has_inf = false;
+
+              auto flattened = tensor.view({-1});
+              for (int64_t i = 0; i < flattened.numel(); i++) {
+                if(std::isinf(flattened[i].item<float>())) {
+                  has_inf = true;
+                  break;
+                }
+              }
+
+              if (assertion) {
+                assert(!has_inf);
+              }
+
+              return has_inf;
+            }
+
+            static inline bool check_has_nan(at::Tensor tensor, bool assertion) {
+              bool has_nan = false;
+
+              auto flattened = tensor.view({-1});
+              for (int64_t i = 0; i < flattened.numel(); i++) {
+                if(std::isnan(flattened[i].item<float>())) {
+                  has_nan = true;
+                  break;
+                }
+              }
+
+              if (assertion) {
+                assert(!has_nan);
```

Comment:
I think it would be useful to report the offending indices if possible (not sure what's possible in our C++ infra)

Reply:
I added a TODO at [line 1515](https://github.com/pytorch/pytorch/pull/109526/files#diff-18714c66f0b1f90de1bfd784bae5fab68a4e54d817556fefa5cdee2621cb06d0R1515). We are still settling with how log would look like (likely c10 logging would be removed)